From e47136865211a887a26989a58d029aa9f38c55c6 Mon Sep 17 00:00:00 2001
From: Vasiliy Stelmachenok <cabopust@yandex.ru>
Date: Tue, 4 May 2021 16:14:30 +0300
Subject: [PATCH] Fixing compatibility with the RT (Real-Time) kernel

Corrects the compatibility of NVIDIA driver modules with the RT kernel.
---
 kernel/conftest.sh | 4 ++--
 kernel/dkms.conf   | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/kernel/conftest.sh b/kernel/conftest.sh
index 0be67cf..a0559c2 100755
--- a/kernel/conftest.sh
+++ b/kernel/conftest.sh
@@ -5267,9 +5267,9 @@ case "$5" in
         fi
 
         if test_configuration_option CONFIG_PREEMPT_RT; then
-            PREEMPT_RT_PRESENT=1
+            PREEMPT_RT_PRESENT=0
         elif test_configuration_option CONFIG_PREEMPT_RT_FULL; then
-            PREEMPT_RT_PRESENT=1
+            PREEMPT_RT_PRESENT=0
         fi
 
         if [ "$PREEMPT_RT_PRESENT" != "0" ]; then
diff --git a/kernel/dkms.conf b/kernel/dkms.conf
index 4381d45..b3299be 100644
--- a/kernel/dkms.conf
+++ b/kernel/dkms.conf
@@ -6,7 +6,7 @@ AUTOINSTALL="yes"
 # this will cause the kernel module build to infer that it was invoked via
 # Kbuild directly instead of DKMS. The dkms(8) manual page recommends quoting
 # the 'make' command name to suppress this behavior.
-MAKE[0]="'make' -j`nproc` NV_EXCLUDE_BUILD_MODULES='__EXCLUDE_MODULES' KERNEL_UNAME=${kernelver} IGNORE_CC_MISMATCH='__IGNORE_CC_MISMATCH' modules"
+MAKE[0]="'make' -j`nproc` IGNORE_PREEMPT_RT_PRESENCE=1 NV_EXCLUDE_BUILD_MODULES='__EXCLUDE_MODULES' KERNEL_UNAME=${kernelver} IGNORE_CC_MISMATCH='__IGNORE_CC_MISMATCH' modules"
 
 # The list of kernel modules will be generated by nvidia-installer at runtime.
 BUILT_MODULE_NAME[0]="nvidia"
-- 
2.31.1

